File: 프로그래머스/0/181844. 배열의 원소 삭제하기/배열의 원소 삭제하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>


int* solution(int arr[], size_t arr_len, int delete_list[], size_t delete_list_len) {
    int i, j, len = 0;
    int* answer = (int*)malloc(arr_len * sizeof(int));
    for(i = 0; i<arr_len; i++){
        for(j = 0; j<delete_list_len; j++){
            if(arr[i] == delete_list[j]) break;
        }
        if(j == delete_list_len) answer[len++] = arr[i];
    }
    return answer;
}

File: 프로그래머스/0/181844. 배열의 원소 삭제하기/배열의 원소 삭제하기.java
import java.util.*;

class Solution {
    public boolean eq(int[] a, int b){
        for(int j : a){
            if(j == b) 
                return true;     
        } return false;  
    }
    
    public int[] solution(int[] arr, int[] delete_list) {     
        ArrayList<Integer> list = new ArrayList<>();
        
        for(int i: arr){
            if(!eq(delete_list, i)){
                list.add(i);
            }       
        }
        
        int[] answer = new int[list.size()];
        int len = 0;
        
        for(int j: list){
            answer[len++] = j;
        }   
        return answer;
    }
    
}
